packages/components/src/drop-zone/provider.js

```diff
@@ -25,17 +25,22 @@ class DropZoneProvider extends Component {
 		// Event listeners
 		this.onDragOver = this.onDragOver.bind( this );
 		this.onDrop = this.onDrop.bind( this );
+		// Context methods so this component can receive data from consumers
+		this.addDropZone = this.addDropZone.bind( this );
+		this.removeDropZone = this.removeDropZone.bind( this );
 		// Utility methods
 		this.resetDragState = this.resetDragState.bind( this );
 		this.toggleDraggingOverDocument = throttle( this.toggleDraggingOverDocument.bind( this ), 200 );
 		this.isWithinZoneBounds = this.isWithinZoneBounds.bind( this );
 
+		this.dropzones = [];
 		this.state = {
+			addDropZone: this.addDropZone,
+			removeDropZone: this.removeDropZone,
```

Comment:
Why are we storing these in state? seems useless?

Reply:
I lifted this object to the constructor so consumers can check whether the context value has changed. If we created the object within the render method, every time the component rendered, we'll be sending the consumers a new object.

Reply:
Agreed with that. In the previous commits the object was the "state" which was worrying me. It was refactored later.